src/device-state.ts

```diff
@@ -514,11 +515,28 @@ export async function getStatus(): Promise<DeviceStatus> {
 		local: {},
 		dependent: {},
 	};
-	theState.local = { ...theState.local, ...currentVolatile };
+	theState.local = {
+		...theState.local,
+		...currentVolatile,
+	};
 	theState.local!.apps = appsStatus.local;
 	theState.dependent!.apps = appsStatus.dependent;
-	if (appsStatus.commit && !applyInProgress) {
-		theState.local!.is_on__commit = appsStatus.commit;
+
+	// Multi-app warning!
+	// If we have more than one app, simply return the first commit.
+	// Fortunately this won't become a problem until we have system apps, and then
+	// at that point we can filter non-system apps leaving a single user app.
+	// After this, for true multi-app, we will need to report our status back in a
+	// different way, meaning this function will no longer be needed
+	const appIds = Object.keys(theState.local!.apps).map((strId) =>
+		parseInt(strId, 10),
+	);
+
+	const appId: number | undefined = appIds[0];
+	const commit = await commitStore.getCommitForApp(appId);
+
+	if (commit != null && !applyInProgress) {
```

Comment:
We had a user asking about this on the forums the other day. If by any chance they poll the device endpoint while apply is in progress, the commit does not show in the property list. Not sure if that is what we want.